internal/resource/url.go

```diff
@@ -422,14 +423,53 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 
 	sess.Config.Region = aws.String(region)
```

Comment:
In the ARN case, I think the above code passes an empty bucket name to `GetBucketRegion`, which seems incorrect.

More broadly, the `arn` case currently matches the behavior of the `s3` case by assuming the same AWS partition as the one we're currently running in (or `aws` if running outside of AWS).  That seems unintuitive, since the partition is already specified in the ARN.  Let's use the partition from the ARN, provided that there's a way to do it.

Reply:
I copied the get region logic under the url scheme cases so that the `s3` case gets the region just as before, and the `arn` case uses the partition from the ARN.